From 4c952fc5e1de9f48f7391617920cf745cc469c17 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 22 Nov 2021 17:38:56 +0000
Subject: [PATCH] GCC: fix initializater list in image_bitmap.cc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC fails like this:
../../third_party/blink/renderer/core/imagebitmap/image_bitmap.cc:409:51:
 error: converting to ‘const blink::IntRect’ from initializer list would
 use explicit constructor ‘constexpr blink::IntRect::IntRect(const gfx::Rect&)’
               409 |   const IntRect& src_rect{parsed_options.crop_rect};
                   |                                                   ^
---
 third_party/blink/renderer/core/imagebitmap/image_bitmap.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/imagebitmap/image_bitmap.cc b/third_party/blink/renderer/core/imagebitmap/image_bitmap.cc
index b404bad..4c3df6f 100644
--- a/third_party/blink/renderer/core/imagebitmap/image_bitmap.cc
+++ b/third_party/blink/renderer/core/imagebitmap/image_bitmap.cc
@@ -406,7 +406,7 @@ static scoped_refptr<StaticBitmapImage> CropImageAndApplyColorSpaceConversion(
   DCHECK(!image->HasData());
 
   IntRect img_rect(gfx::Point(), IntSize(image->width(), image->height()));
-  const IntRect& src_rect{parsed_options.crop_rect};
+  const IntRect& src_rect = IntRect(parsed_options.crop_rect);
   const IntRect intersect_rect = IntersectRects(img_rect, src_rect);
 
   // If cropRect doesn't intersect the source image, return a transparent black
@@ -664,7 +664,7 @@ ImageBitmap::ImageBitmap(ImageData* data,
   if (DstBufferSizeHasOverflow(parsed_options))
     return;
 
-  const IntRect& src_rect{parsed_options.crop_rect};
+  const IntRect& src_rect = IntRect(parsed_options.crop_rect);
   const IntRect data_src_rect = IntRect(gfx::Point(), data->Size());
   const IntRect intersect_rect =
       crop_rect ? IntersectRects(src_rect, data_src_rect) : data_src_rect;
-- 
2.32.0

